.github/actions/check_version/get_version_from_files.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+import os, sys
+import argparse
+import struct
+
+
+#
+# MARK: - argparse
+#
+
+
+parser = argparse.ArgumentParser(
+    description="Check os version compliance with standards used in the project"
+)
+
+parser.add_argument("firmware_hex_path", type=str, help="path of firmware hex file")
+parser.add_argument("os_bin_path", type=str, help="path of os binary file")
+
+args = parser.parse_args()
+
+
+#
+# MARK: - Functions
+#
+
+
+def addToEnvironement(data: str):
+    env_file = os.getenv("GITHUB_ENV")
+    with open(env_file, "a") as env:
+        env.write(data + "\n")
+
+
+def getDataFromFile(file_path: str, mode="r") -> bytes:
+    data: bytes
+    try:
+        f = open(file_path, mode)
+        data = f.read()
+        f.close()
+    except BaseException as error:
+        print(f"{error}")
+        sys.exit(1)
+
+    return data
```

Comment:
following the previous comment, this can be extracted in another file as well and imported where needed